### PR TITLE
[9.0][MIG] Upgrade medical to 9.0.

### PR DESCRIPTION
**Pull request by woodbrettm**
Man this diff is absolutely massive. The OCA commit history goes up to Jan 30 2016, and any commits between there and now on release/9.0 are all laslabs ones which is nice. Means I can just copy paste the folder.

**Couple things to bring to light:**
- Are the file headers for python and xml correct? Will be using that header on all files for now on if good.
- Is this [help](https://github.com/laslabs/vertical-medical/compare/9.0...laslabs:custom/9.0/LABS-287-medical-v9?expand=1#diff-0180c4fc21f7260dcce8093bf66149ceR47) attribute correct for is_work?
- Does laslabs/9.0 share the same config files as OCA/9.0? I noticed the travis build had less build jobs than usual (compared to release/9.0), so not sure if the travis config file needs to be improved to add stuff.

### CODE REVIEW

File: medical/__openerp__.py [hunk 1]
```diff
@@ -1,72 +1,30 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Comment (lasley) [inline medical/__openerp__.py]: Remove our perpetual copyright, also I accidentally deleted some of the original attributions - we will need to fix that.

```
Copyright 2004-2009 Tiny SPRL
Copyright 2016 LasLabs Inc.
```

File: medical/README.rst [hunk 2]
```diff
@@ -6,82 +6,64 @@
 Odoo Medical
 ============
 
-What is Odoo Medical
---------------------
+This module extends Odoo with the base functionality of medical patients, centers, and doctors.
```
Comment (lasley) [inline medical/README.rst]: This module does not provide doctors anymore.

File: medical/README.rst [hunk 3]
```diff
@@ -6,82 +6,64 @@
...
 
-**Odoo Medical** is a multi-user, highly scalable, centralized Electronic Medical
-Record (EMR) and Hospital Information System for odoo.
+**Some patient attributes available:**
```
Comment (lasley) [inline medical/README.rst]: While we can leave it, it's not necessary to list attributes provided. Concept of patient and medical centers does it. If anything, explain what a medical center or patient is, not what their data looks like.

File: medical/README.rst [hunk 4]
```diff
@@ -6,82 +6,64 @@
...
-
 Bug Tracker
 ===========
 
 Bugs are tracked on `GitHub Issues <https://github.com/OCA/vertical-medical/issues>`_.
 In case of trouble, please check there if your issue has already been reported.
 If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
-`here <https://github.com/OCA/vertical-medical/issues/new?body=module:%20medical%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
-
+`here <https://github.com/OCA/vertical-medical/issues/new?body=module:%20medical%0Aversion:%209.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (lasley) [inline medical/README.rst]: Change this to bug tracker blurb to current template

File: medical/__init__.py [hunk 5]
```diff
@@ -1,25 +1,6 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Comment (lasley) [inline medical/__init__.py]: Remove perpetual copyrihgt, add 2004 from Tech-Receptives

File: medical/__openerp__.py [hunk 6]
```diff
@@ -1,72 +1,30 @@
...
-    'category': 'Medical',
-    'depends': [
-        'base',
-        'product',
+    "name": "Odoo Medical",
+    "summary": "Extends Odoo with medical patients, doctors, and centers.",
+    "version": "9.0.1.0.0",
+    "category": "Medical",
+    "website": "https://odoo-community.org/",
+    "author": "LasLabs, Odoo Medical Team, Odoo Community Association (OCA)",
```
Comment (lasley) [inline medical/__openerp__.py]: Remove `Odoo Medical Team` - they're not a thing anymore

File: medical/__openerp__.py [hunk 7]
```diff
@@ -1,72 +1,30 @@
...
+
 {
-    'name': 'Odoo Medical',
-    'version': '8.0.1.1.0',
-    'category': 'Medical',
-    'depends': [
-        'base',
-        'product',
+    "name": "Odoo Medical",
+    "summary": "Extends Odoo with medical patients, doctors, and centers.",
```
Comment (lasley) [inline medical/__openerp__.py]: No doctors

File: medical/data/ir_sequence_data.xml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Comment (lasley) [inline medical/data/ir_sequence_data.xml]: Remove perpetual copyright

File: medical/demo/medical_patient_demo.xml [hunk 9]
```diff
@@ -0,0 +1,58 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Comment (lasley) [inline medical/demo/medical_patient_demo.xml]: Perpetual copyright

File: medical/demo/medical_patient_demo.xml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+
+<odoo>
+
+	<record id="res_partner_patient_1" model="res.partner">
+		<field name="name">Emma Fields</field>
```
Comment (lasley) [inline medical/demo/medical_patient_demo.xml:8]: Add ref https://fakena.me/fake-social-security-number/

File: medical/demo/medical_patient_demo.xml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+		<field name="parent_id" ref="base.partner_demo" />
+	</record>
+
+	<record id="medical_patient_patient_1" model="medical.patient">
+		<field name="partner_id" ref="medical.res_partner_patient_1" />
+		<field name="dob">1920-02-23</field>
+	</record>
+	
+	<record id="res_partner_patient_2" model="res.partner">
+		<field name="name">Kyle Jones</field>
```
Comment (lasley) [inline medical/demo/medical_patient_demo.xml:27]: Add ref https://fakena.me/fake-social-security-number/

File: medical/demo/medical_patient_demo.xml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+		<field name="parent_id" ref="base.partner_demo" />
+	</record>
+	
+	<record id="medical_patient_patient_2" model="medical.patient">
+		<field name="partner_id" ref="medical.res_partner_patient_2" />
+		<field name="dob">1950-01-02</field>
+	</record>
+	
+	<record id="res_partner_patient_3" model="res.partner">
+		<field name="name">Gary Smith</field>
```
Comment (lasley) [inline medical/demo/medical_patient_demo.xml:46]: Add ref https://fakena.me/fake-social-security-number/

File: medical/demo/medical_patient_demo.xml [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+		<field name="zip">89121</field>
+		<field name="street">123 Test Drive</field>
+		<field name="email">kyle@example.com</field>
+		<field name="phone">+1 (702) 343-2347</field>
+		<field name="parent_id" ref="base.partner_demo" />
+	</record>
+	
+	<record id="medical_patient_patient_3" model="medical.patient">
+		<field name="partner_id" ref="medical.res_partner_patient_3" />
+		<field name="dob">1980-11-02</field>
```
Comment (lasley) [inline medical/demo/medical_patient_demo.xml:60]: let's kill this patient - add `dod`

File: medical/medical_constants.py [hunk 14]
```diff
@@ -1,24 +1,6 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Comment (lasley) [inline medical/medical_constants.py]: Perpetual copyright, add tech-receptives

File: medical/models/__init__.py [hunk 15]
```diff
@@ -1,28 +1,6 @@
...
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Comment (lasley) [inline medical/models/__init__.py]: Perpetual copyright, add tech-receptives

File: medical/models/medical_patient.py [hunk 16]
```diff
@@ -1,105 +1,108 @@
...
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU Affero General Public License for more details.
-#
-# You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-# #############################################################################
-from openerp import models, fields, api
+# Copyright 2016-TODAY LasLabs Inc. (https://laslabs.com)
```
Comment (lasley) [inline medical/models/medical_patient.py]: Perpetual copyright, add tech-receptives

File: medical/models/medical_patient.py [hunk 17]
```diff
@@ -1,105 +1,108 @@
...
+                    deceased = ''
+                years_months_days = '%s%s %s%s %s%s%s' % (
+                    delta.years, _('y'), delta.months, _('m'),
+                    delta.days, _('d'), deceased
+                )
+            else:
+                years_months_days = _('No DoB !')
+            rec_id.age = years_months_days
+
+    @api.constrains('is_pregnant', 'gender')
```
Comment (lasley) [inline medical/models/medical_patient.py:94]: Add `api.multi`

**Review by lasley — CHANGES_REQUESTED** [review]

**Comment by lasley** [comment]
@bmw95 - dunno about `is_work` - just leave it, will probably remove in v10. Don't worry about anything outside of the module dir.

File: medical/__openerp__.py (hunk 1 above)
Comment (woodbrettm) [reply medical/__openerp__.py]: Was actually me ;). Wasn't quite sure how to go about that so know now for future.

Comment (woodbrettm) [reply medical/__openerp__.py]: Done

File: medical/__openerp__.py (hunk 7 above)
Comment (woodbrettm) [reply medical/__openerp__.py]: Done

File: medical/__openerp__.py (hunk 6 above)
Comment (woodbrettm) [reply medical/__openerp__.py]: Done

File: medical/README.rst (hunk 2 above)
Comment (woodbrettm) [reply medical/README.rst]: Done

File: medical/README.rst (hunk 3 above)
Comment (woodbrettm) [reply medical/README.rst]: Reduced to "This module extends Odoo with the base functionality of medical patients and centers." Removed attribute descriptions.

File: medical/README.rst (hunk 4 above)
Comment (woodbrettm) [reply medical/README.rst]: Done

File: medical/__init__.py (hunk 5 above)
Comment (woodbrettm) [reply medical/__init__.py]: Done

File: medical/data/ir_sequence_data.xml (hunk 8 above)
Comment (woodbrettm) [reply medical/data/ir_sequence_data.xml]: Done

File: medical/demo/medical_patient_demo.xml (hunk 9 above)
Comment (woodbrettm) [reply medical/demo/medical_patient_demo.xml]: Done

File: medical/demo/medical_patient_demo.xml (hunk 13 above)
Comment (woodbrettm) [reply medical/demo/medical_patient_demo.xml:60]: Done

File: medical/medical_constants.py (hunk 14 above)
Comment (woodbrettm) [reply medical/medical_constants.py]: Done

File: medical/models/__init__.py (hunk 15 above)
Comment (woodbrettm) [reply medical/models/__init__.py]: Done

File: medical/models/medical_patient.py (hunk 16 above)
Comment (woodbrettm) [reply medical/models/medical_patient.py]: Done

File: medical/models/medical_patient.py (hunk 17 above)
Comment (woodbrettm) [reply medical/models/medical_patient.py:94]: Done. multi placed first then constrains.

File: medical/demo/medical_patient_demo.xml (hunk 10 above)
Comment (woodbrettm) [reply medical/demo/medical_patient_demo.xml:8]: Done

File: medical/demo/medical_patient_demo.xml (hunk 11 above)
Comment (woodbrettm) [reply medical/demo/medical_patient_demo.xml:27]: Done

File: medical/demo/medical_patient_demo.xml (hunk 12 above)
Comment (woodbrettm) [reply medical/demo/medical_patient_demo.xml:46]: Done

**Comment by woodbrettm** [comment]
### Changes Made In Addition to Review Requests
-   Add gender to demo patients
-   Refactor tests to use demo data (so much faster now!)
-   Add patient compute_deceased method (with depends on dod)
-   Add tests that cover the following cases:
  - Ensure pregnant female doesn’t raise any ValidationErrors
  - Split age test into 3 separate tests (no dob, deceased, and normal cases)
  - Test deceased is True if dod
  - Test deceased is False if no dod
  - Refactored action_invalidate test

### Thoughts
- Should there be language translations on strings used in the tests if they can be translated in the core logic (e.g. validationerror messages, etc)?
- Was curious if we should add deceased field to the patient view with logic to allow it to be editable. Would allow for rare cases where the patient is deceased but no dod has been confirmed yet.

**Comment by woodbrettm** [comment]
Forgot to change manifest website key to laslabs domain.

**Review by lasley — APPROVED** [review]
This looks good to me, thanks @bmw95 

Regarding your questions:
- Don't worry about the translations, non-issue
- There will never be an instance in the medical field of a non-confirmed date of death. If they are marked as dead, a date will be known

**Pull request merged**